packages/firestore/src/local/memory_persistence.ts

```diff
@@ -180,6 +180,7 @@ export class MemoryPersistence implements Persistence {
     this.referenceDelegate.onTransactionStarted();
     return transactionOperation(txn)
       .next(result => {
+        txn.raiseOnCommittedEvent();
```

Comment:
This ordering seems off. Couldn't the reference delegate conceivably modify the transaction?

IndexedDbPersistence has the better implementation. Chaining this off the promise makes it clearer that it's altogether outside the transaction.

Reply:
Changed this to invoke the committed handler as the last step.